### ISSUE TITLE
Document -Force parameter for Publish-AzWebApp

### ISSUE THREAD

**Issue by okpedersen** [Azure/azure-powershell#10224]
## Description of the new feature

The `-Force` argument should be documented for `Publish-AzWebApp`. 

(Unsure of whether this is the correct type of issue, but none of the other seemed to fit.)

**Comment by msJinLei** [comment]
@okpedersen 
Could you provide the your enviroment information?

## Environment data
Please run $PSVersionTable and paste the output in the below code block

## Module versions
Please run (Get-Module  -ListAvailable) and paste the output in the below code block

We dont' have the -Force parameter for this cmdlet currently and so what do you mean?

**Comment by okpedersen** [comment]
Our Azure DevOps release pipeline fails with an error message if the `-Force` parameter is not supplied to `Publish-AzWebApp`. 

The error message: 
`Windows PowerShell is in NonInteractive mode. Read and Prompt functionality is not available.`

Also, the source code in `PublishAzureWebApp.cs` ([here](https://github.com/Azure/azure-powershell/blob/master/src/Websites/Websites/Cmdlets/WebApps/PublishAzureWebApp.cs#L42-43) and [here](https://github.com/Azure/azure-powershell/blob/master/src/Websites/Websites/Cmdlets/WebApps/PublishAzureWebApp.cs#L102)) seems to indicate support for `-Force`? 

# Environment data

```
Name                           Value                                                                                   
----                           -----                                                                                   
PSVersion                      5.1.14393.3053                                                                          
PSEdition                      Desktop                                                                                 
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
BuildVersion                   10.0.14393.3053                                                                         
CLRVersion                     4.0.30319.42000                                                                         
WSManStackVersion              3.0                                                                                     
PSRemotingProtocolVersion      2.3                                                                                     
SerializationVersion           1.1.0.1 
```

# Module versions

I'm unsure of whether I can paste this information, due to the corporate environment I'm in. I could probably send this privately if absolutely necessary.

However, some information: We're running Azure PowerShell 4.* (preview) in a AzureDevOps release pipeline, and are currently running the Az-module and AzureRM-module side-by-side.

**Comment by msJinLei** [comment]
> However, some information: We're running Azure PowerShell 4.* (preview) in a AzureDevOps release pipeline, and are currently running the Az-module and AzureRM-module side-by-side.

This part of information is really important. Anyway we shall get to know what the versions of Az and Az.websites you are using so that we can reproduce the issue.

**Comment by okpedersen** [comment]
> This part of information is really important. 

Yes, we are aware that it is not recommended to run both versions in parallel, but do not have any other options currently.

> Are you sure you are using the Az 4.x Preview?

No, I see that I wasn't clear enough. We run an Azure PowerShell 4.* (preview) **task** in an Azure DevOps release pipeline. See [here](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-powershell?view=azure-devops). The YAML for our task looks like this:

```
steps:
- task: AzurePowerShell@4
  displayName: 'Azure PowerShell script: FilePath'
  inputs:
    azureSubscription: '<subscription name>'
    ScriptPath: '$(System.DefaultWorkingDirectory)/$(Release.PrimaryArtifactSourceAlias)/artifacts/pipeline.ps1'
    ScriptArguments: '-Target deploy -Environment dev'
    azurePowerShellVersion: LatestVersion
```

> We shall get to know what the versions of Az and Az.websites you are using so that we can reproduce the issue. 

All installed Az and AzureRM modules:

```
Script     1.6.2      Az.Accounts                         
Script     1.0.1      Az.Advisor                          
Script     1.0.2      Az.Aks                              
Script     1.1.1      Az.AnalysisServices                 
Script     1.3.1      Az.ApiManagement                    
Script     1.0.2      Az.ApplicationInsights              
Script     1.3.3      Az.Automation                       
Script     1.1.1      Az.Batch                            
Script     1.0.1      Az.Billing                          
Script     1.3.1      Az.Cdn                              
Script     1.2.1      Az.CognitiveServices                
Script     2.6.0      Az.Compute                          
Script     1.0.1      Az.ContainerInstance                
Script     1.1.0      Az.ContainerRegistry                
Script     1.3.0      Az.DataFactory                      
Script     1.0.1      Az.DataLakeAnalytics                
Script     1.2.2      Az.DataLakeStore                    
Script     1.0.1      Az.DeploymentManager                
Script     1.0.0      Az.DevTestLabs                      
Script     1.1.1      Az.Dns                              
Script     1.2.2      Az.EventGrid                        
Script     1.4.0      Az.EventHub                         
Script     1.1.1      Az.FrontDoor                        
Script     2.0.2      Az.HDInsight                        
Script     1.3.0      Az.IotHub                           
Script     1.3.1      Az.KeyVault                         
Script     1.3.1      Az.LogicApp                         
Script     1.1.1      Az.MachineLearning                  
Script     1.0.1      Az.ManagedServices                  
Script     1.0.1      Az.MarketplaceOrdering              
Script     1.1.0      Az.Media                            
Script     1.3.0      Az.Monitor                          
Script     1.14.0     Az.Network                          
Script     1.1.0      Az.NotificationHubs                 
Script     1.3.3      Az.OperationalInsights              
Script     1.1.3      Az.PolicyInsights                   
Script     1.1.0      Az.PowerBIEmbedded                  
Script     1.4.5      Az.RecoveryServices                 
Script     1.1.0      Az.RedisCache                       
Script     1.0.2      Az.Relay                            
Script     1.7.0      Az.Resources                        
Script     1.4.0      Az.ServiceBus                       
Script     1.2.0      Az.ServiceFabric                    
Script     1.1.0      Az.SignalR                          
Script     1.14.2     Az.Sql                              
Script     1.7.0      Az.Storage                          
Script     1.2.0      Az.StorageSync                      
Script     1.0.0      Az.StreamAnalytics                  
Script     1.0.2      Az.TrafficManager                   
Script     1.4.2      Az.Websites                         
Script     5.1.2      Azure                               
Script     0.5.4      Azure.AnalysisServices              
Manifest   0.5.0      Azure.AnalysisServices              
Script     0.4.2      Azure.AnalysisServices              
Script     4.6.0      Azure.Storage                       
Script     4.2.1      Azure.Storage                       
Script     6.9.0      AzureRM                             
Script     5.7.0      AzureRM                             
Script     0.6.14     AzureRM.AnalysisServices            
Script     0.6.6      AzureRM.AnalysisServices            
Script     6.1.5      AzureRM.ApiManagement               
Script     5.1.2      AzureRM.ApiManagement               
Script     0.1.8      AzureRM.ApplicationInsights         
Script     0.1.3      AzureRM.ApplicationInsights         
Script     5.1.1      AzureRM.Automation                  
Script     4.3.2      AzureRM.Automation                  
Script     4.0.10     AzureRM.Backup                      
Script     4.0.4      AzureRM.Backup                      
Script     4.1.5      AzureRM.Batch                       
Script     4.0.6      AzureRM.Batch                       
Script     0.14.6     AzureRM.Billing                     
Script     0.14.1     AzureRM.Billing                     
Script     5.0.6      AzureRM.Cdn                         
Script     4.2.2      AzureRM.Cdn                         
Script     0.9.10     AzureRM.CognitiveServices           
Script     0.9.4      AzureRM.CognitiveServices           
Script     5.6.0      AzureRM.Compute                     
Script     4.6.0      AzureRM.Compute                     
Script     0.3.7      AzureRM.Consumption                 
Script     0.3.1      AzureRM.Consumption                 
Script     0.2.10     AzureRM.ContainerInstance           
Script     0.2.5      AzureRM.ContainerInstance           
Script     1.0.10     AzureRM.ContainerRegistry           
Script     1.0.4      AzureRM.ContainerRegistry           
Script     5.0.3      AzureRM.DataFactories               
Script     4.2.2      AzureRM.DataFactories               
Script     0.5.10     AzureRM.DataFactoryV2               
Script     0.5.3      AzureRM.DataFactoryV2               
Script     5.1.4      AzureRM.DataLakeAnalytics           
Script     4.2.3      AzureRM.DataLakeAnalytics           
Script     6.1.2      AzureRM.DataLakeStore               
Script     5.2.0      AzureRM.DataLakeStore               
Script     4.0.9      AzureRM.DevTestLabs                 
Script     4.0.4      AzureRM.DevTestLabs                 
Script     5.1.0      AzureRM.Dns                         
Script     4.1.2      AzureRM.Dns                         
Script     0.3.7      AzureRM.EventGrid                   
Script     0.3.2      AzureRM.EventGrid                   
Script     0.6.10     AzureRM.EventHub                    
Script     0.6.3      AzureRM.EventHub                    
Script     4.1.8      AzureRM.HDInsight                   
Script     4.1.2      AzureRM.HDInsight                   
Script     5.1.4      AzureRM.Insights                    
Script     4.0.4      AzureRM.Insights                    
Script     3.1.8      AzureRM.IotHub                      
Script     3.1.2      AzureRM.IotHub                      
Script     5.2.1      AzureRM.KeyVault                    
Script     4.3.0      AzureRM.KeyVault                    
Script     4.1.4      AzureRM.LogicApp                    
Script     4.0.3      AzureRM.LogicApp                    
Script     0.18.5     AzureRM.MachineLearning             
Script     0.17.2     AzureRM.MachineLearning             
Script     0.4.8      AzureRM.MachineLearningCompute      
Script     0.4.2      AzureRM.MachineLearningCompute      
Script     0.2.5      AzureRM.MarketplaceOrdering         
Script     0.2.1      AzureRM.MarketplaceOrdering         
Script     0.10.4     AzureRM.Media                       
Script     0.9.2      AzureRM.Media                       
Script     6.8.0      AzureRM.Network                     
Script     5.4.2      AzureRM.Network                     
Script     5.0.3      AzureRM.NotificationHubs            
Script     4.1.1      AzureRM.NotificationHubs            
Script     5.0.6      AzureRM.OperationalInsights         
Script     4.3.2      AzureRM.OperationalInsights         
Script     1.0.4      AzureRM.PolicyInsights              
Script     4.1.10     AzureRM.PowerBIEmbedded             
Script     4.1.4      AzureRM.PowerBIEmbedded             
Script     5.6.0      AzureRM.Profile                     
Script     4.6.0      AzureRM.Profile                     
Script     4.1.8      AzureRM.RecoveryServices            
Script     4.1.2      AzureRM.RecoveryServices            
Script     4.4.1      AzureRM.RecoveryServices.Backup     
Script     4.1.2      AzureRM.RecoveryServices.Backup     
Script     0.2.10     AzureRM.RecoveryServices.SiteRec... 
Script     0.2.4      AzureRM.RecoveryServices.SiteRec... 
Script     5.0.6      AzureRM.RedisCache                  
Script     4.1.2      AzureRM.RedisCache                  
Script     0.3.10     AzureRM.Relay                       
Script     0.3.3      AzureRM.Relay                       
Script     6.5.0      AzureRM.Resources                   
Script     5.5.2      AzureRM.Resources                   
Script     0.16.10    AzureRM.Scheduler                   
Script     0.16.3     AzureRM.Scheduler                   
Script     4.1.2      AzureRM.ServerManagement            
Script     0.6.12     AzureRM.ServiceBus                  
Script     0.6.4      AzureRM.ServiceBus                  
Script     0.3.12     AzureRM.ServiceFabric               
Script     0.3.4      AzureRM.ServiceFabric               
Script     1.0.0      AzureRM.SignalR                     
Script     5.0.6      AzureRM.SiteRecovery                
Script     4.11.3     AzureRM.Sql                         
Script     4.4.0      AzureRM.Sql                         
Script     5.1.0      AzureRM.Storage                     
Script     4.2.3      AzureRM.Storage                     
Script     4.0.10     AzureRM.StreamAnalytics             
Script     4.0.4      AzureRM.StreamAnalytics             
Script     4.0.5      AzureRM.Tags                        
Script     4.0.1      AzureRM.Tags                        
Script     4.1.1      AzureRM.TrafficManager              
Script     4.0.3      AzureRM.TrafficManager              
Script     4.0.5      AzureRM.UsageAggregates             
Script     4.0.2      AzureRM.UsageAggregates             
Script     5.1.0      AzureRM.Websites                    
Script     4.2.2      AzureRM.Websites                        
```

**Comment by ThejaChoudary** [comment]
Taking a look at this..

**Comment by panchagnula** [comment]
@markcowl is there any reason -Force parameter wouldn't work from Devops pipeline.
@ThejaChoudary tried this from cloudshell & local and we are able to successfully use this parameter. Thanks!

**Comment by panchagnula** [comment]
@ThejaChoudary the documentation see [link](https://docs.microsoft.com/en-us/powershell/module/az.websites/publish-azwebapp?view=azps-3.5.0) doesn't have Force documented so we should just update the help to show this. Most likely caused because the help was hand-crafted instead of generated

**Comment by markcowl** [comment]
@panchagnula @ThejaChoudary Correect.  Actually regenerating the help should automatically show the parameter

**Comment by panchagnula** [comment]
@ThejaChoudary/ @Kotasudhakarreddy did you make a fix for this already and so can you add a link to this PR here and close?

**Comment by ThejaChoudary** [comment]
Fixed by PR [#11196](https://github.com/Azure/azure-powershell/pull/11196)
